trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -2435,16 +2529,25 @@ private Statement createReachabilityAssert(final ILocation loc, final String fun
 		// https://github.com/sosy-lab/sv-benchmarks/pull/1001
 		final Check check;
 		if (checkProperty) {
-			final Function<Spec, String> funPosMessage =
-					s -> s == Spec.ERROR_FUNCTION ? "call to " + functionName + " is unreachable"
-							: Check.getDefaultPositiveMessage(s);
-			final Function<Spec, String> funNegMessage =
-					s -> s == Spec.ERROR_FUNCTION ? "a call to " + functionName + " is reachable"
-							: Check.getDefaultNegativeMessage(s);
+			final CheckMessageProvider posMsgProvider = new CheckPositiveMessageProvider();
+			final CheckMessageProvider negMsgProvider = new CheckNegativeMessageProvider();
+
+			/* overwrite result message for reachable error functions */
+			posMsgProvider.registerMessageOverride(Spec.ERROR_FUNCTION,
+					() -> String.format("a call to %s is unreachable", functionName));
+			negMsgProvider.registerMessageOverride(Spec.ERROR_FUNCTION,
+					() -> String.format("a call to %s is reachable", functionName));
+
+			/* overwrite result message for assertions with error messages */
+			if (errorMsg != null && !errorMsg.isEmpty()) {
+				negMsgProvider.registerMessageOverride(Spec.ASSERT,

```

Comment:
Do we only want to override the message for asserts (for now this is of course true 😉) or could you just override the message for `spec`?

Reply:
Each specific `Check` should implement its own message customization. For example, a `ReqCheck` defines other messages for a `Spec` than a regular `Check`.

Reply:
Yes, but I mean just here. Wouldn't it be better to to use `negMsgProvider.registerMessageOverride(spec,...)` instead of `negMsgProvider.registerMessageOverride(Spec.ASSERT,...)` here? Not crucial, should just allow us to have messages for sth. else than assertions here.

Reply:
No you can't do that, because each `Spec` requires its own result message. Since the function `createReachabilityAssert(...)` defines behavior for various `Spec`s, we have to distinguish the type of `Spec` here.

Note that default messages are automatically used as fallback messages, i.e. if there is no overwrite for a specific `Spec` type is defined, the default message is used.